### PR TITLE
Extract virtual buildeps apk set

### PR DESCRIPTION
**Pull request by ahawkins**

### CODE REVIEW

**Comment by ahawkins** [comment]
@terlar I think you know more about Alpine linux than I do. I tried to shrink this image more by following the patterns in the official images. See https://github.com/docker-library/ruby/blob/bfc7a48724ceb1917ddbcb713b24c835eca584c8/2.1/alpine/Dockerfile for example. I read the `APKBUILD` file for each apk installed to extract the build dependencies. I created the `.await-buildeps` virtual package separately, added the various apks required for the util, then removed the virtual installs. Curiously this did not make any change in image size compared to the image built in #1. If you're knowledgeable about this stuff, can you check it and see if something is up? The `apk del` says it removed ~180mb of files but that's not reflected in the image size.

**Comment by terlar** [comment]
```
$ docker images | grep await
saltside/await  latest 1593e8499f11  9 minutes ago  96.16 MB
```

When removing the heavy mongodb dependency we are down to 96MB.

I banged my head against this for quite a while until I realized it was all the mongodb package. See the package reference here https://pkgs.alpinelinux.org/package/edge/testing/x86_64/mongodb. That is 110MB.

**Review by terlar — APPROVED** [review]
With my changes all should be good, what do you think?

**Comment by 0dayfall** [comment]
Approved ;-)

File: bin/await [hunk 1]
```diff
@@ -76,7 +76,7 @@ main() {
 			shift $((OPTIND-1))
 
 			while [ "${counter}" -lt "${retry}" ]; do
-				if mongo --quiet "${host}:${port}/local" --eval 'db.stats()' > /dev/null ; then
+				if mongostats "mongodb://${host}:${port}/local" > /dev/null ; then
```
Comment (ahawkins) [inline bin/await]: Can you add in the `$@` so end users can forward more options to `mongostats`?

**Review by ahawkins — COMMENTED** [review]
Can you address my comment about `$@`? Users should be able to use `await foo -- [FOO_OPTIONS]` for everything.

**Comment by ahawkins** [comment]
> With my changes all should be good, what do you think?

@terlar looks great. 98MB. Not bad :D Can you address my line note regarding UX?

**Pull request merged**